### ISSUE TITLE
GNOME Shell 3.20.x JS Type Error

### ISSUE THREAD

**Issue by nobeh** [lucasdiedrich/gnome-display-switcher#12]
Running on GNOME Shell 3.20.x on Ubuntu 16.04:

```
(gnome-shell:2330): Gjs-WARNING **: JS ERROR: TypeError: this._scrollView.hscroll is null
SwitcherList<.highlight@resource:///org/gnome/shell/ui/switcherPopup.js:421
wrapper@resource:///org/gnome/gjs/modules/lang.js:178
SwitcherPopup<._select@resource:///org/gnome/shell/ui/switcherPopup.js:294
wrapper@resource:///org/gnome/gjs/modules/lang.js:178
SwitcherPopup<._initialSelection@resource:///org/gnome/shell/ui/switcherPopup.js:109
wrapper@resource:///org/gnome/gjs/modules/lang.js:178
SwitcherPopup<.show@resource:///org/gnome/shell/ui/switcherPopup.js:141
wrapper@resource:///org/gnome/gjs/modules/lang.js:178
SwitcherManager<._show@/home/nobakht/.local/share/gnome-shell/extensions/display-switcher@lucas.diedrich.gmail.com/switcherUI.js:47
wrapper@resource:///org/gnome/gjs/modules/lang.js:178
DisplayExtension<._show@/home/nobakht/.local/share/gnome-shell/extensions/display-switcher@lucas.diedrich.gmail.com/extension.js:53
wrapper@resource:///org/gnome/gjs/modules/lang.js:178
```
- First time the widget shows the selector menu for displays but does not do anything.
- Second time, the above error appears

**Comment by lucasdiedrich** [comment]
This is related to something on Ubuntu 16.04, i've never tested this extension on Gnome 3.20, i added the support because one Arch user said that it works on his workstation. 

Anyway, i'm updating my Fedora's version to 24 and i would be able to test that.

**Comment by lucasdiedrich** [comment]
So, i've tested on Fedora 24 with Gnome 3.20.x and its working. So with Arch and Fedora works, must be something with Ubuntu 16.04 not been able to bind the keys correctly, or, the app is missing something after you first select anything.

Can you please execute de following command, i try to reproduce the error again? 
`sudo journalctl /usr/bin/gnome-session -f`

If its something with the app, it probably gonna show some interesting log there.

Thanks.

**Comment by nobeh** [comment]
Thanks for looking into this. The log I provided above was captured by:

```bash
$ journalctl /usr/bin/gnome-shell -f -o cat
```

I was not able to capture anything from `gnome-session`. 

----

On a fresh GNOME session, the first time I press super + I:

```
(gnome-shell:2318): Gjs-WARNING **: JS ERROR: TypeError: ival[2] is undefined
DisplayHandler<._parse@/home/nobakht/.local/share/gnome-shell/extensions/display-switcher@lucas.diedrich.gmail.com/displayHandler.js:152
wrapper@resource:///org/gnome/gjs/modules/lang.js:178
DisplayHandler<._reload@/home/nobakht/.local/share/gnome-shell/extensions/display-switcher@lucas.diedrich.gmail.com/displayHandler.js:170
wrapper@resource:///org/gnome/gjs/modules/lang.js:178
DisplayHandler<._getMode@/home/nobakht/.local/share/gnome-shell/extensions/display-switcher@lucas.diedrich.gmail.com/displayHandler.js:84
wrapper@resource:///org/gnome/gjs/modules/lang.js:178
DisplayHandler<._getIndex@/home/nobakht/.local/share/gnome-shell/extensions/display-switcher@lucas.diedrich.gmail.com/displayHandler.js:113
wrapper@resource:///org/gnome/gjs/modules/lang.js:178
SwitcherManager<._show@/home/nobakht/.local/share/gnome-shell/extensions/display-switcher@lucas.diedrich.gmail.com/switcherUI.js:48
wrapper@resource:///org/gnome/gjs/modules/lang.js:178
DisplayExtension<._show@/home/nobakht/.local/share/gnome-shell/extensions/display-switcher@lucas.diedrich.gmail.com/extension.js:53
wrapper@resource:///org/gnome/gjs/modules/lang.js:178
```

and then selecting any option does not seem to do anything. Note `ival[2] is undefined`.

And, then after the first time, I get this:

```
(gnome-shell:2318): Gjs-WARNING **: JS ERROR: TypeError: this._scrollView.hscroll is null
SwitcherList<.highlight@resource:///org/gnome/shell/ui/switcherPopup.js:421
wrapper@resource:///org/gnome/gjs/modules/lang.js:178
SwitcherPopup<._select@resource:///org/gnome/shell/ui/switcherPopup.js:294
wrapper@resource:///org/gnome/gjs/modules/lang.js:178
SwitcherPopup<._initialSelection@resource:///org/gnome/shell/ui/switcherPopup.js:109
wrapper@resource:///org/gnome/gjs/modules/lang.js:178
SwitcherPopup<.show@resource:///org/gnome/shell/ui/switcherPopup.js:141
wrapper@resource:///org/gnome/gjs/modules/lang.js:178
SwitcherManager<._show@/home/nobakht/.local/share/gnome-shell/extensions/display-switcher@lucas.diedrich.gmail.com/switcherUI.js:47
wrapper@resource:///org/gnome/gjs/modules/lang.js:178
DisplayExtension<._show@/home/nobakht/.local/share/gnome-shell/extensions/display-switcher@lucas.diedrich.gmail.com/extension.js:53
wrapper@resource:///org/gnome/gjs/modules/lang.js:178
```

What is the "Laptop mode" in the settings? It is set to `false` for me and I'm using a laptop with dual displays.

HTH.

**Comment by lucasdiedrich** [comment]
So thats the case, you're not the first person complaining about this, but i have never recieved enough information to track this.

**Laptop mode** is enabled on laptops where we've to predefine the onboard always as the primary graphic card, if its false it is going to work as a desktop and some times this can break some stuff. Can you please run the command below and paste return for me?

`$ xrandr`

**Comment by nobeh** [comment]
Unfortunately, I do not have access to my external displays till Monday. I ran `xrandr` with and without `--verbose` switch. Both attached. HTH. 
[xrander.txt](https://github.com/lucasdiedrich/gnome-display-switcher/files/366793/xrander.txt)
[xrander-verbose.txt](https://github.com/lucasdiedrich/gnome-display-switcher/files/366794/xrander-verbose.txt)

**Comment by lucasdiedrich** [comment]
That's odd, eDP1 is yours internal laptop display and that's is exactly what i use to set the **laptop-mode** on. The unique diference is that i user the following command to get that:

> $ xrandr --current

Can you check if the return is the same as running just _xrandr_, it should be.

**Comment by nobeh** [comment]
Thanks for the time on this. It seems that `xrandr` and `xrandr --curent` creates the same output for me. The file attached.

[xrander-current.txt](https://github.com/lucasdiedrich/gnome-display-switcher/files/368033/xrander-current.txt)

**Comment by lucasdiedrich** [comment]
This extension will be disabled since this functionalty is available at Gnome 3.26+. To use Gnome just press 'Super' + P and hold Super.
